botnim/query.py

```diff
@@ -111,20 +112,97 @@ def get_index_mapping(self) -> Dict:
             logger.error(f"Failed to get index mapping: {str(e)}")
             raise
 
-def run_query(query_text: str, environment: str, bot_name: str, context_name: str, num_results: int = 7) -> List[SearchResult]:
+    def get_default_num_results(self) -> int:
+        """
+        Get the default number of results for the current context
+        
+        Returns:
+            int: Default number of results
+        """
+        # Find the specific context configuration
+        context_config = next(
+            (ctx for ctx in self.config.get('context', []) if ctx['name'] == self.context_name),
+            {}
+        )
+        
+        # Get default num_results from context config or use a reasonable default
+        return context_config.get('default_num_results', 7)
+
+    def query(self, query_text: str, num_results: int = None) -> List[SearchResult]:
```

Comment:
Why do we need this thin wrapper around `search`?

Reply:
in a way it is a left over of a previous version (which you asked me to fix). The only usage was to better deal with num_results. I moved it into the search function.